src/update_ram.c

```diff
@@ -64,9 +75,13 @@ void RAMFUNCTION wolfBoot_start(void)
     }
 
     for (;;) {
-        if ((wolfBoot_open_image(&os_image, active) < 0) ||
-            (wolfBoot_verify_integrity(&os_image) < 0) ||
-            (wolfBoot_verify_authenticity(&os_image) < 0)) {
+        if (((ret[0] = wolfBoot_open_image(&os_image, active)) < 0) ||
```

Comment:
Not sure this is needed. The status of each of the three operation is kept in `os_image.hdr_ok`, `os_image.sha_ok` and `os_image.signature_ok`, respectively, which can be used in the xil_printf below. Assignments in conditions are more prone to errors.

Reply:
It is the return code from `wolfBoot_verify_signature` being captured here. I've cleaned this up.